### PR TITLE
1.0.0

### PR DESCRIPTION
**Pull request by derekrushforth**
Hey @matt-west,

No rush on getting to this. It's a huge review and I know you have a lot on your plate right now. This PR introduces a load of improvements to the template designs and workflow. Since there are breaking changes I've upgraded MailMason to version 1.0.0. Here's a rundown on the changes:

## General improvements
- Upgraded a bunch of packages as well as the node version (LTS 10.16.0)
- Styles are compiled to `dist/stylesheets` folder instead of the `src/stylesheets`

## Design improvements
- Added 2 new layouts. You might have seen these with our layouts release. Layouts also come with their own global stylesheet so they can be themed.
- Each template now uses [Nunito Sans](https://fonts.google.com/specimen/Nunito+Sans) as the main typeface. I’ve also added a fallback CSS class to each layout that inserts a compatible font stack for old versions of Outlook. Email
- Support dark/high contrast mode. This uses the `prefers-color-scheme: dark` media query. It’s more of a progressive enhancement for email clients that support it. Email-dark

## Accessibility
- Background and text colors now pass AA and AAA contrast ratios defined by WCAG 2.0
- Added role=“presentation” to presentational tables so that they are not unnecessarily read by screen readers

## Introduce template pushing
I restructured the templates folder so that it will work with our CLI tool. You’ll now see that each template is in its own folder with the content and metadata JSON. 

~One thing to note is that this _only_ pushes the templates, not layouts. I don’t think it’s that helpful to upload the layouts to Postmark if someone is already using a workflow like MailMason. Plus, we’d have to do a bunch of extra work to our grunt build so that the Handlebars layouts are compatible with Postmark’s mustachio syntax. I don’t think it’s worth it right now.~

Edit: I changed my mind :) I needed a system where I could easily build and drop these files into our app's starter templates. Without that, it's a lot of manual work. So I figured it was worth doing now.

There are two main types of template builds going on in MailMason now: a generic workflow that compiles layouts/templates into a single file and a Postmark-specific workflow that spits out separate templates and layouts.  Here's an example of the new `dist` folder structure:
```
dist
├── compiled
│   └── welcome
│       ├── content.html
│       └── content.txt
├── postmark-layouts
│   └── plain
│       ├── content.html
│       ├── content.txt
│       └── meta.json
├── postmark-templates
│   └── welcome
│       ├── content.html
│       ├── content.txt
│       └── meta.json
└── stylesheets
    ├── basic-full.css
    ├── basic.css
    └── plain.css
```

Templates under `compiled` contain the templates and layouts combined into one HTML and Text version. The goal is that these templates are mainly for customers who plan on using a different ESP.
- No `meta.json` with these templates since these aren't intended to be pushed to Postmark 
- CSS is inlined using Premailer by default. Prior to this, we relied on special tasks(`grunt testBuild`) to inline the CSS since it was assumed that Postmark would do the inlining.
- These templates are used for generating previews in `previews.html`

The Postmark-specific stuff is located under `postmark-layouts` and `postmark-templates`.
-  Each template and layout contain a `meta.json` that is copied from the `src` folder so that users can push them to Postmark
- Handlebar's `{{body}}` tag in the layouts is converted to Mustachio's `{{{@content}}}` placeholder
- CSS is loaded into the style block in each layout. It is not inlined since Postmark takes care of that.

This obviously complicates the Grunt configuration a lot 😬. Especially for non-Postmark users. I carefully renamed a bunch of Grunt tasks to include "postmark" in the name so that people can rip these tasks out if they only plan on generating generic templates. I made note of this in Gruntfile.js, but I will also mention it in the wiki.

cc @rianvdm

### CODE REVIEW

**Comment by matt-west** [comment]
@derekrushforth At a quick glance, this looks awesome. I’ll set aside some time early next week to review all the code.

File: Gruntfile.js [hunk 1]
```diff
@@ -231,100 +273,170 @@ module.exports = function(grunt) {
...
-  grunt.registerTask('spam', ['spamcheck']);
-  grunt.registerTask('litmus', ['testBuild', 'postmark:litmus']);
-  grunt.registerTask('flood', ['testBuild', 'postmark:flood']);
-
-  // Before sending tests via Postmark, ensure that test builds with inlined CSS are generated
-  grunt.registerTask('testBuild', ['default', 'copy:testTemplates', 'premailer:html']);
-};
+  grunt.registerTask('spam', ['spamcheck'])
+  grunt.registerTask('litmus', ['default', 'postmark:litmus'])
+  grunt.registerTask('flood', ['default', 'postmark:flood'])
```
Comment (matt-west) [inline Gruntfile.js]: When I run this I receive the compiled emails but also get a bunch of emails with just the body content, but no layout. I think it’s just sending all of the html files in the list folder, but should we only be sending the compiled templates?

File: src/stylesheets/basic-full.scss [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+/*Media Queries ------------------------------ */
+
+@media only screen and (max-width: 600px) {
+  .email-body_inner,
+  .email-footer {
+    width: 100% !important;
+  }
+}
+
+@media (prefers-color-scheme: dark) {
```
Comment (matt-west) [inline src/stylesheets/basic-full.scss]: The footer background color needs updating when in dark mode.

Screenshot 2019-07-15 at 10 21 40

**Comment by derekrushforth** [comment]
Thanks @matt-west! I fixed the issues that you brought up.


I definitely think that putting together an upgrade guide is a good idea. It might be a little tricky since the entire folder structure changed. And especially if someone modified their grunt tasks. I'll do my best to put something together in the release notes.

**Comment by derekrushforth** [comment]
@matt-west If you're happy with the fixes can you approve this? I'll also be working on some release notes and an upgrade guide.

**Pull request merged**